lib/IRGen/IRGenMangler.cpp

```diff
@@ -103,7 +103,8 @@ IRGenMangler::withSymbolicReferences(IRGenModule &IGM,
       // when the referent may be in another file, once the on-disk
       // ObjectMemoryReader can handle them.
       // Private entities are known to be accessible.
-      if (type->getEffectiveAccess() >= AccessLevel::Internal &&
+      if (type->getEffectiveAccess(true /*assume private import disabled*/) >=
```

Comment:
I'm guessing you need this because the runtime cannot realize types with a textual mangling of the private discriminator?

We already emit private discriminators for type metadata for debug info. So I wonder if we can fix this bug by teaching the runtime to use that information in more cases?

Reply:
> I'm guessing you need this because the runtime cannot realize types with a textual mangling of the private discriminator?
Yes.

Reply:
`type->getFormalAccessScope().isPublic()` does not give the right answer for e.g ``Innermost``. It will answer with true.

```
fileprivate struct Foo {
  fileprivate struct Inner: P {
    fileprivate struct Innermost { }
    typealias A = Innermost
  }
}```

Reply:
That should not happen, since we use getFormalAccessScope() for all kinds of checking; I would expect other problems if its wrong. @jrose-apple Can you take a look?

Reply:
I am passing nullptr.

Reply:
Let me step thru this again.

Reply:
:( I had the condition inverted.